server/server.lua

```diff
@@ -207,6 +199,15 @@ RegisterNetEvent('luke_garages:SpawnVehicle', function(model, plate, coords, hea
                         end
                     end
                 end
+                local playerPed = GetPlayerPed(xPlayer.source)
+                local timer = GetGameTimer()
+                while GetVehiclePedIsIn(playerPed) ~= entity do
+                    Wait(0)
+                    SetPedIntoVehicle(playerPed, entity, -1)
+                    if timer - GetGameTimer() > 15000 then
+                        break
+                    end
+                end
```

Comment:
I'd definitely prefer if the ped *wasn't* being teleported into the vehicle

Reply:
In my testing it was necessary. the race condition was usually caused by players logging in and out around the garage. i guess it was swapping owners. being the driver forces you to become the owner. i can try it again without that. but this has been working for me.

Reply:
> In my testing it was necessary. the race condition was usually caused by players logging in and out around the garage. i guess it was swapping owners. being the driver forces you to become the owner. i can try it again without that. but this has been working for me.

It just feels really out of place for the resource imo, especially since you have to by design get out of the vehicle to store it.

Reply:
i understand. i'll try it without the putting the ped in the vehicle. maybe the other changes were enough to solve it.